src/Faker/Provider/Internet.php

```diff
@@ -102,7 +102,7 @@ public function userName()
         $format = static::randomElement(static::$userNameFormats);
         $username = static::bothify($this->generator->parse($format));
 
-        $username = strtolower(static::transliterate($username));
+        $username = static::toLower(static::transliterate($username));
```

Comment:
@krsriq 

These changes more tightly couple this provider to the class it extends.

Perhaps it makes sense to keep this as is, in the light of #15?

Reply:
So a PR that does the opposite - remove the use of all the `static::toUpper` and `static::toLower` - would probably make more sense?
E.g. in
https://github.com/FakerPHP/Faker/blob/afa2aab00ddf99b57a028f7f86347e5526c80569/src/Faker/Provider/Address.php#L85

and in all the localized providers
https://github.com/FakerPHP/Faker/blob/afa2aab00ddf99b57a028f7f86347e5526c80569/src/Faker/Provider/da_DK/Address.php#L245

So how would we call `Base` methods in 2.0? Should e.g. `Base::toUpper` be called directly (instead of via `static::`)?

Implementing the `mbstring` check of  [`Base::toUpper`](https://github.com/FakerPHP/Faker/blob/afa2aab00ddf99b57a028f7f86347e5526c80569/src/Faker/Base.php#531) in each provider would introduce quite a bit of duplication.

Reply:
I dont really see added value of these methods at all tbh. Php can handle this on its own without proxying